torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -154,10 +143,11 @@ def record_post_forward(self, handle: Optional[FlatParamHandle]) -> None:
         if not handle:
             return
         # Only record the first usage of a handles key
-        if handle in self.handles_to_post_forward_order_index:
```

Comment:
![image](https://github.com/pytorch/pytorch/assets/4755252/d5616115-280e-4b1a-be7d-1b53687b4986)

How it looked before, in the reverted PR. That would cause a None to show up.